python/diff_frames.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+import os
+import sys
+import argh
+import asyncio
+import subprocess
+from collections import namedtuple
+
+import pysndfile
+
+from terminal import LinePrintingTerminal, Keyboard
+
+bdiff_frames_path = os.path.join(
+    os.path.dirname(__file__), os.pardir, 'diff_frames')
+
+Hunk = namedtuple('Hunk', ('start_a', 'end_a', 'start_b', 'end_b'))
+
+
+def duration(sndfile):
+    return sndfile.frames() / sndfile.samplerate()
+
+
+def fmt_seconds(t):
+    hours, remainder = divmod(t, 3600)
+    minutes, seconds = divmod(remainder, 60)
+    if not hours:
+        if not minutes:
+            return '{:.2f}'.format(seconds)
+        return '{:.0f}:{:.1f}'.format(minutes, seconds)
+    return '{:.0f}:{:.0f}:{.0f}'.format(hours, minutes, seconds)
+
+
+class DiffApp:
+    def __init__(self, filename_a, filename_b, loop=None):
+        self.filename_a = filename_a
+        self.filename_b = filename_b
+        self._psf_a = None
+        self._psf_b = None
+        self._diff = None
+        self._len = None
+
+        self._loop = loop or asyncio.get_event_loop()
```

Comment:
Yes, it was eventually so that when writing tests we could do that, but you're right, there's not much point in doing it like that and not for everything else. I was actually thinking of making the loop internal to the App class for now, just because there's no need to expose it currently!